nautobot_netbox_importer/generator/source.py

```diff
@@ -417,6 +418,19 @@ def __str__(self) -> str:
         """Return a string representation of the wrapper."""
         return f"{self.__class__.__name__}<{self.content_type} -> {self.nautobot.content_type}>"
 
+    def cache_record_ids(self, source: RecordData) -> None:
+        """Cache record identifier mappings."""
+        uid = self.get_pk_from_data(source)
+
+        pk_uid = source.get(self.nautobot.pk_field.name, None)
+        if pk_uid:
+            self._uid_to_pk_cache[pk_uid] = uid
+
+        if self.identifiers and len(self.identifiers) == 1:
+            identifier_uid = source.get(self.identifiers[0], None)
+            if identifier_uid:
+                self._uid_to_pk_cache[identifier_uid] = uid
```

Comment:
Can you add a bit more context in the docstring? I don't quite understand why there are *three* different uids involved, I would have thought there'd just be two of them (netbox/nautobot).